File: tests/node/stack_overflow.js
// Stack overflow - NO signal expected (usually)
// V8 detects stack overflow via guard pages internally and throws a
// RangeError: Maximum call stack size exceeded.
// This is handled WITHIN V8 - the SIGSEGV on the guard page is caught by
// V8's own signal handler and converted to a JS exception.
// crash-tracer may see a transient SIGSEGV that V8 handles - this is a
// key FALSE POSITIVE scenario.

console.error("[node/stack_overflow] Recursing until V8 stack limit...");

function recurse(n) {
    return recurse(n + 1);
}

try {
    recurse(0);
} catch (e) {
    console.error(`[node/stack_overflow] V8 caught it as JS exception: ${e.message}`);
    console.error("[node/stack_overflow] Process survived - no real crash occurred");
}

File: tests/java/StackOverflow.java
// Stack overflow — NO signal expected
// The JVM detects stack overflow via guard pages internally and throws
// StackOverflowError, a normal Java exception. The SIGSEGV on the guard
// page is caught by the JVM's own signal handler and converted to the
// Java exception. crash-tracer may see a transient SIGSEGV — this is a
// key FALSE POSITIVE scenario (same as V8/Node).

public class StackOverflow {
    public static void main(String[] args) {
        System.err.println("[java/stack_overflow] Recursing until JVM stack limit...");

        try {
            recurse(0);
        } catch (StackOverflowError e) {
            System.err.println("[java/stack_overflow] JVM caught it as StackOverflowError");
            System.err.println("[java/stack_overflow] Process survived - no real crash occurred");
        }
    }

    static int recurse(int n) {
        return recurse(n + 1);
    }
}
